src/item_location.cpp

```diff
@@ -17,35 +17,126 @@
 
 #include <climits>
 
-class item_location::impl
+template <typename T>
+static unsigned do_pack( const T &sel, const item *obj )
```

Comment:
Naming, find_index() and get_index()?  Because pack and unpack don't tell me anything.